test/unit/requester-core.test.js

```diff
@@ -79,6 +79,47 @@ describe('requester util', function () {
             });
         });
 
+        it('should accept URL in uppercase', function () {
```

Comment:
Also, test for other capitalization combination.

Reply:
@codenirvana I think that would be redundant since they are handled differently in the function, so as per the current tests those cases should already be covered.

Reply:
I mean something like `Https`, `HttP`.

Reply:
@codenirvana wouldn't that be redundant too, testing an inbuilt JS function ?? when we know the flag is there and we have a valid regex which ignores case altogether.